sdk/src/sdk/vdb/grpc/kuksa_val_v2/BrokerAsyncGrpcFacade.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2022-2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "BrokerAsyncGrpcFacade.h"
+
+#include "sdk/Logger.h"
+#include "sdk/grpc/GrpcCall.h"
+
+#include <grpcpp/channel.h>
+
+namespace velocitas::kuksa_val_v2 {
+
+BrokerAsyncGrpcFacade::BrokerAsyncGrpcFacade(std::shared_ptr<grpc::Channel> channel)
+    : m_stub{kuksa::val::v2::VAL::NewStub(channel)} {}
+
+void BrokerAsyncGrpcFacade::GetValues(
+    kuksa::val::v2::GetValuesRequest&&                                  request,
+    std::function<void(const kuksa::val::v2::GetValuesResponse& reply)> replyHandler,
+    std::function<void(const grpc::Status& status)>                     errorHandler) {
```

Comment:
If `replyHandler` and `errorHandler` are complex might be expensive to pass them by value, maybe you could better pass these as const references?

Reply:
It's not really passing the function itself, but only a reference/pointer to it. This is the newer approach replacing the "old-school" passing of function pointers.

Reply:
Still the question is, if it's possible to pass a std::function object via reference? I pretty sure, that  the answer is finally "no": Passing a function object is often passing a lambda function that is often defined in the local context of a function. This local context will be destroyed when the function is left. But the passed function object does need a longer lifetime because it's a callback.
Maybe it could make sense to use move semantics here ...